hash-range-store/s3.tf

```diff
@@ -2,4 +2,14 @@ resource "aws_s3_bucket" "bucket" {
   bucket = local.bucket_name
 
   tags = var.tags
+
+  lifecycle_rule {
+    id      = "objects_to_standard_ia"
+    enabled = var.cycle_objects_to_standard_ia
+
+    transition {
+      days          = 30
+      storage_class = "STANDARD_IA"
+    }
```

Comment:
Assuming we then get charged less for storage, but more for access?

I ask as things like reindexes will definitely be run on old objects, and a lot of them, which might not save us money.

Reply:
Yup, although only on objects >128KB. Objects smaller than that (which I imagine is the vast majority of the catalogue) won't be cycled to Standard-IA, because the overhead for S-IA is actually greater.

You need to be looking up every object that gets cycled at least once a month before the access cost exceeds the storage cost (although it's fractions of a cent, honestly).

Where this helps us is in the long tail of VHS content that's no longer being accessed.

Reply:
This is mostly for the benefit of the storage service, where the storage manifests bucket is 40GB of Standard IA content and growing, and most of it is never accessed after it's written.

Reply:
Ah, I must have missed the 128kb, is that just an AWS thing?

Reply:
Yeah, from https://docs.aws.amazon.com/AmazonS3/latest/dev/storage-class-intro.html:

> The S3 Intelligent-Tiering storage class is suitable for objects larger than 128 KB that you plan to store for at least 30 days. **If the size of an object is less than 128 KB, it is not eligible for auto-tiering.** Smaller objects can be stored, but they are always charged at the frequent access tier rates in the S3 Intelligent-Tiering storage class.